### PR TITLE
Fix for Docker Image

### PR DESCRIPTION
**Pull request by lawrencegripper**
The current docker image crashes on start as it doesn't restore dependencies for the individual '/server' and '/client' directories. Only the root 'package.json' is copied into the container at the time 'npm install' is run. 

Fix:

Added create the '/server' and '/client' directories to container with their package.json files
Added lock files to keep things in sync
Moved to using yarn command in dockerfile
Added build step for '/client' directory to rebuild src (ts and sass) assets as part of docker build

### CODE REVIEW

**Comment by torosent** [comment]
Docker build is failing. See attached output.
[TerminalOutput.txt](https://github.com/CatalystCode/ibex-dashboard/files/1189481/TerminalOutput.txt)

Now that [Issue 16766](https://github.com/npm/npm/issues/16766) & [Issue 16892](https://github.com/npm/npm/issues/16892) are closed, we can use node 8.2.1 with Web Apps on linux 👍

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,33 @@
-FROM node:6.11-alpine
+FROM node:8.2.1-alpine
```
Comment (torosent) [inline Dockerfile]: Now that [Issue 16766](https://github.com/npm/npm/issues/16766) & [Issue 16892](https://github.com/npm/npm/issues/16892) are closed, we can use node 8.2.1 with Web Apps on linux 👍

File: Dockerfile [hunk 2]
```diff
@@ -1,15 +1,33 @@
...
 RUN npm install yarn -g
-RUN npm install
+RUN yarn
 
 # Bundle app source
 COPY . /usr/src/app
 
+# Build client assets
+WORKDIR /usr/src/app/client
+RUN yarn build
```
Comment (torosent) [inline Dockerfile]: Build is failing

See attached output
[TerminalOutput.txt](https://github.com/CatalystCode/ibex-dashboard/files/1189490/TerminalOutput.txt)

File: Dockerfile (hunk 1 above)
Comment (lawrencegripper) [inline Dockerfile]: Sorry forgot I updated the node version

**Pull request merged**